libs/core-components/src/lib/Button/Button.tsx

```diff
@@ -57,7 +60,8 @@ const Button = ({
   // Manage prop warnings
   /* eslint-disable no-console */
   if (type === 'link' && size) {
-    size = '';
+    // DISABLING: empty string is not a valid value for size
+    // size = '';
```

Comment:
I think I have a solution. Please see:

https://github.com/TACC/tup-ui/compare/task/TUP-272--core-components-fixup...task/TUP-272--core-components-fixup--suggest-intrinsic-width

I tested similar change on Portal (not comitted). UI did not break. Tests did not fail.

Reply:
Let's discuss further. Ideally, there should be a way to avoid generating a console warning through better type design even if that only applies for TypeScript based clients of this library.

Reply:
I think I found the solution:
- https://dev.to/msholtyfd/using-typescript-to-guard-against-specific-react-prop-combinations-3c9a
- https://www.youtube.com/watch?v=yqn9Fkv7f2M

I've tried it out in ac5dcf8, so we can review it in https://github.com/TACC/tup-ui/pull/14.